### ISSUE TITLE
Bazel combined cache not working with GCP Storage

### ISSUE THREAD

**Issue by pauldraper** [bazelbuild/bazel#15993]
### Description of the bug:

As of 5.2.0, Bazel doesn't download from GCP storage when the disk cache is used.

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

**WORKSPACE.bazel**

```bzl
```

**BUILD.bazel**

```bzl
genrule(
  name = "example",
  cmd = "date; echo example > $@",
  outs = ["example.txt"],
)
```

**.bazelrc**

```txt
build --disk_cache=~/.cache/bazel-disk
build --google_default_credentials
build --remote_cache=https://storage.googleapis.com/example-bucket
```

Build:

```sh
bazel build :example
```

Then clean:

```sh
bazel clean
rm -r ~/.cache/bazel-disk/*
```

Then build again:

```sh
bazel build :example
```

The output for the last step shows: `2 processes: 1 internal, 1 linux-sandbox.`

It ought to display: `2 processes: 1 remote cache hit, 1 internal.`

### Which operating system are you running Bazel on?

Ubuntu 20.04

### What is the output of `bazel info release`?

release 5.2.0

### Any other information, logs, or outputs that you want to share?

Oddly, this works as expected if I do any one of the following:

1. Remove the `--disk_cache` option.
2. Use a local HTTP server (nginx) instead of GCP Storage.
3. Use Bazel 5.1.0.

**Comment by tjgq** [comment]
It appears that this was introduced in 4d900ceea but later fixed in a24d1bc30d. As a result, it made it into the 5.2 release, but both 5.3 and 6.0 won't have the bug; I've confirmed that the provided test case doesn't repro at HEAD, nor in the 5.3 release branch.

**Comment by pauldraper** [comment]
Excellent!

Now we just need a 5.3.0 release 😅

**Comment by pauldraper** [comment]
While I can believe that 4d900ce broke it, I am skeptical that a23d1bc fixed it.

**Comment by fmeum** [comment]
Maybe they meant https://github.com/bazelbuild/bazel/commit/4d900ceea12919ad62012830a95e51f9ec1a48bb?

**Comment by tjgq** [comment]
Sorry, copy paste error. I meant it was fixed by 40e485dc67 (originally https://github.com/bazelbuild/bazel/pull/15842).
